TeamCharles/initial-site
# 138-productIsActive - Using Product.IsActive

## Status
**READY**

## Documentation

[ ] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[X] This PR does **NOT** require documentation.

## Description
Changed the behavior of deleting a product. Products are now marked as `IsActive = false`. Products are never deleted from the database now. This is needed if we ever need to view past orders and a seller decides to stop selling a product.

## Resolves Issue Number

https://github.com/TeamCharles/initial-site/issues/138

## Deploy Notes

```sh
git pull --prune
git checkout 138-productIsActive
dotnet run
```

## Steps to Test or Reproduce

1. Log into an account

2. Add a product to your cart

3. Log out

4. Log into the account that is selling that original product

5. Find a product you are selling

6. Hit the "Delete" button next to the product to stop selling it.

7. Try to find it in the View All Products view

8. Try to find it in the Category view

9. Log back into first account

10. View your Cart (Unavailable item should be noted in your cart)

11. Confirm the order (Unavailable item should not be included in the Confirmation page)

## Impacted Files in Application
List general components of the application that this PR will affect:

* ProductsController

* CartController

* ProductTypeView

* ProductSubTypeView

* Base View Model
